### ISSUE TITLE
handle undef in FileEditor _is_reference_newer

### ISSUE THREAD

**Issue by stdweird** [quattor/CAF#160]
```
[VERB] Opening file /etc/sysconfig/network
Use of uninitialized value $msg[1] in join or string at /usr/lib/perl/CAF/Reporter.pm line 491.
[DEBUG] Reference file (<undef>) older than /etc/sysconfig/network
Use of uninitialized value $_[2] in join or string at /usr/lib/perl/CAF/Reporter.pm line 457.
[DEBUG] Reading initial contents from /etc/sysconfig/network
```

**Comment by stdweird** [comment]
maybe merge #154 first?

**Comment by jouvin** [comment]
These 2 messages seem to be fixed by #154. But I found 2 other similar messages:
```
Use of uninitialized value $args[1] in join or string at /exp/si/jouvin/GitRepositories/Quattor/CAF/target/dependency/build-scripts/Test/Quattor/Object.pm line 169.
```
One of them is triggered by `test-caffileeditor_add_or_replace.t`.

**Comment by stdweird** [comment]
`Test::Quattor::Object` is from maven-tools, but the issue i saw was not in testing, but in actual usage. i'll open a PR to fix it
